doc/api/assert.md

```diff
@@ -1164,6 +1164,9 @@ changes:
 Tests strict equality between the `actual` and `expected` parameters as
 determined by the [SameValue Comparison][].
 
+[`assert.deepStrictEqual()`][] should be used for all objects and all
+arrays, not only nested ones.
```

Comment:
This is not really true as it depends on the use case. If someone wants to verify reference equality of objects, it is indeed the right function to use. No matter the shape of the object.
And arrays are always objects, so distinguishing them does not seem correct to me either.

Reply:
To me, `deep` hints that there is some nesting involved. I'm hoping that it could be better accentuated that it's not the case here.

> And arrays are always objects, so distinguishing them does not seem correct to me either.
True, it's only distinguished in JSON and it might be confusing here so I'll remove it.

Reply:
@bessbd Right, that’s why @BridgeAR is correct in saying that this is *not* the right thing for all use cases and we shouldn’t be recommending using it for all objects.

Reply:
> deep hints that there is some nesting involved.

The shape is not an important factor when it comes to the difference between the comparisons used. They might be used for any shape and it's exactly what a user requires.

What is important, is the referenced `SameValue Comparison` algorithm. It describes how the algorithm itself defines the equality.

A `deep` equality normally means that each property will recursively be inspected and compared. This also applies to the top level and "flat" objects.
The "regular" equality check is for primitive values and reference equality, just as described in the mentioned algorithm. It is not about "flat" objects.

Reply:
Then how about  a 

    NB: for comparing objects (including arrays) based on value instead of reference, consider using [`assert.deepStrictEqual()`][] regardless of nestedness.
?

Reply:
I'd say "contents" not "value", and i'd add something about "recursively" for "deep".